project/Build.scala

```diff
@@ -1244,7 +1244,12 @@ object Build {
       // TODO add versions etc.
       def srcManaged(v: String, s: String) = s"out/bootstrap/stdlib-bootstrapped/scala-$v/src_managed/main/$s-library-src"
       def scalaSrcLink(v: String, s: String) = s"-source-links:$s=github://scala/scala/v$v#src/library"
-      def dottySrcLink(v: String, s: String) = s"-source-links:$s=github://lampepfl/dotty/$v#library/src"
+      def dottySrcLink(v: String, s: String) =
+        sys.env.get("GITHUB_SHA") match {
```

Comment:
Why do we need this `environment variables` driven logic?

Reply:
The problem is that we want to link to the source code that has been actually used for this PR, not the 3.0.0 sources. Before we tried to link to e.g. experimental.scala that that not exists in 3.0.0